src/app/shared/image/image-loader.component.ts

```diff
@@ -1,21 +1,42 @@
 import { Component, Input, ElementRef } from '@angular/core';
+import { HalDoc } from '../cms/haldoc';
 
 @Component({
   selector: 'image-loader',
   styleUrls: ['image-loader.component.css'],
   template: `
-    <img *ngIf="src" [src]="src" class="loading" (load)="onLoad()"/>
+    <img *ngIf="src" [src]="src" (load)="onLoad()" (error)="onError()"/>
     `
 })
 
 export class ImageLoaderComponent {
 
   @Input() public src: string;
+  @Input() public imageDoc: HalDoc;
+
+  static PLACEHOLDER_IMAGE: string = '/assets/images/placeholder.png';
+  static PLACEHOLDER_ERROR: string = '/assets/images/placeholder_error.png';
 
   constructor(private element: ElementRef) {}
 
-  onLoad() {
-    this.element.nativeElement.style['background-image'] = `url(${this.src})`;
+  setBackgroundImage(src) {
+    this.element.nativeElement.style['background-image'] = `url(${src})`;
+  }
+
+  onLoad = () => this.setBackgroundImage(this.src);
+
+  onError = () => this.setBackgroundImage(ImageLoaderComponent.PLACEHOLDER_ERROR);
+
+  ngOnChanges(changes:any) {
```

Comment:
Almost seems like a bug on their side, given [the doc example](https://angular.io/docs/ts/latest/api/core/index/OnChanges-class.html).  Yeah, may be worth adding that just in case they fix it someday.